### ISSUE TITLE
Build Errors fixed by using atom-runner?

### ISSUE THREAD

**Issue by logista** [Arcath/jekyll-atom#33]
**tl;dr:**  `atom-runner` does something to my Atom environment that gets lost when it's taken away, and `jekyll` uses whatever it is. I don't think this is necessarily the fault of `jekyll` but I'm not sure where else to report it.

Caveat: I'm no coder, but if you have questions I'll try my best to answer them

- System: Mac OS X 10.10.4 [XCode not installed, though it may have been at one time]
- Atom: Version 1.0.2
- Jekyll: 1.0.0
- Atom-Runner: 2.4.1

1. For some reason I couldn't get `jekyll` to build my site.
2. I installed a script runner package (`atom-runner`) to check a few things from within Atom, and then `jekyll` worked.
3. So I uninstalled `atom-runner` and restarted Atom and `jekyll` stopped working.
4. Reinstalled `atom-runner` and `jekyll` works again

More details:

- Try 1
    - Install `jekyll`
    - Set Build Command to `jekyll, build, --drafts`
    - Start Server
    - Response: `Jekyll Binary Incorrect`
    - Build doesn't work, but server runs
- Try 2
    - Set Build Command to my explicit path (gotten from running `which jekyll` and copying it over)
    - Start Server
    - Response: `Jekyll site build failed!`
    - Build doesn't work, but server runs
- Try 3
    - Set Build Command to my explicit path, with `/gems/` replaced by `/wrappers/` (gotten from Issue #2)
    - Start Server
    - Response: `Jekyll Binary Incorrect`
    - Build doesn't work, but server runs
- Try 4
    - Install `atom-runner`
    - Start Server
    - Response: `Jekyll site build complete!`
    - Everything works, including `--drafts` and build-on-save
- Check
    - Disable `atom-runner`
    - Start Server
    - Response: `Jekyll site build complete!`
    - Everything works, including `--drafts` and build-on-save
    - RESTART ATOM
    - Start Server
    - Response: `Jekyll Binary Incorrect`
    - Reenable `atom-runner`
    - Start Server
    - Response: `Jekyll site build complete!`

**Comment by Arcath** [comment]
Is this still happening with 1.1.1?

**Comment by logista** [comment]
Uh, it's worse now. I had worked around it by starting `atom` from Terminal (instead of Finder), and now `jekyll package` throws an error:

[Enter steps to reproduce below:]

1. Open my jekyll site from Terminal (with and without `atom-runner`).
or
2. Open my jekyll site from Finder (with and without `atom-runner`)

 All packages appear to be up to date

**Atom Version**: 1.4.1
**System**: Mac OS X 10.11.3
**Thrown From**: [jekyll](https://github.com/arcath/jekyll-atom) package, v1.1.2


### Stack Trace

Uncaught TypeError: atom.packages.getActivePackage(...).mainModule.toggleServer is not a function

```
At /Users/barbara/.atom/packages/jekyll/lib/jekyll/toolbar-view.coffee:82

TypeError: atom.packages.getActivePackage(...).mainModule.toggleServer is not a function
    at JekyllManagerView.module.exports.JekyllManagerView.toggleServer (/Users/barbara/.atom/packages/jekyll/lib/jekyll/toolbar-view.coffee:82:66)
    at HTMLButtonElement.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/node_modules/space-pen/lib/space-pen.js:220:36)
    at HTMLButtonElement.jQuery.event.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/node_modules/space-pen/node_modules/jquery/dist/jquery.js:4435:9)
    at HTMLButtonElement.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-space-pen-views/node_modules/space-pen/node_modules/jquery/dist/jquery.js:4121:28)
```

### Commands

```
     -0:03.0 jekyll:toolbar (atom-text-editor.editor.mini.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "markdown-preview",
      "deprecation-cop",
      "exception-reporting",
      "language-c",
      "language-clojure",
      "language-csharp",
      "language-go",
      "language-make",
      "language-mustache",
      "language-objective-c",
      "language-perl",
      "language-python",
      "language-sql",
      "metrics",
      "timecop",
      "welcome",
      "language-pfm",
      "autocomplete-atom-api"
    ],
    "packagesWithKeymapsDisabled": [
      "file-templates"
    ],
    "themes": [
      "one-light-ui",
      "pen-paper-coffee-syntax"
    ]
  },
  "jekyll": {
    "buildCommand": [
      "jekyll",
      "build",
      "--drafts",
      "--future"
    ],
    "draftByDefault": true,
    "postsType": ".md"
  }
}
```

### Installed Packages

```coffee
# User
Sublime-Style-Column-Selection, v1.5.1
autocomplete-bibtex, v0.6.2
autocomplete-emojis, v2.2.2
color-picker, v2.1.0
jekyll, v1.1.2
markdown-preview-plus, v2.2.2
mathjax-wrapper, v0.1.0
pen-paper-coffee-syntax, v0.13.0
pigments, v0.22.0
wordcount, v2.5.0

# Dev
No dev packages
```

**Comment by Arcath** [comment]
I have updated the toolbar view to get the main module in a different way which might help.

Out of interest does it work without the tool bar toggling the server through `Jekyll: Toggle Server` `Alt+Shift+T`?

**Comment by logista** [comment]
Yay! I can run it using `Jekyll: Toggle Server Alt+Shift+T` as long as I start `atom` from Terminal. I don't seem to need `atom-runner` anymore. I think this is probably an OS X problem, so I'm ok with how it works.

Thanks for working on this, I would be lost without this package!

**Comment by herrbischoff** [comment]
I can confirm that I have the same issue. After installing `atom-runner` it appears to work as expected.

**Comment by firewheel** [comment]
Chiming in with @logista that on Mac OS 10.11.3 I'm getting the same `Jekyll site build failed!` error when launching Atom via the Dock, but when launched via Terminal it builds successfully.

However, on an otherwise identical environment using Mac OS 10.11.2, it works just fine regardless of launch method. Going to update to 10.11.4 and see how that one plays.
